resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/localstore/Bucket.java

```diff
@@ -380,6 +380,9 @@ private Map<String, Object> loadEntries(File indexFile) throws CoreException, IO
 				resultEntries.put(readEntryKey(source), readEntryValue(source));
 			}
 			return resultEntries;
+		} catch (RuntimeException e) {
```

Comment:
I would either move this into `load` (line 349) or also catch the IO exception here and let the method only throw a core exception.

Reply:
From runtime it doesn't make a difference. I chose from possible callers that closest to the exception. It already throws such a CoreException from org.eclipse.core.internal.properties.PropertyBucket.readEntryValue(DataInputStream) Line 296, but could also fail with other RuntimeExceptions.
The function loadEntries() is documented to throw IOException and i rather keep the code changes as small as possible to avoid long git history.